### PR TITLE
Add test for pkg/agent/core/ngt/service/vqueue/option

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

### Description:

I added test code for `pkg/agent/core/ngt/serice/vqueue/option.go` and godoc comment.
And I refactored based on a document of option error handling and fixed [ci fails error](https://github.com/vdaas/vald/runs/2596843952).

**gramer check passed**

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/agent/core/ngt/service/vqueue/option.go [hunk 1]
```diff
@@ -31,51 +33,61 @@ var defaultOptions = []Option{
...
+		if eg == nil {
+			return errors.NewErrInvalidOption("errgroup", eg)
 		}
+		v.eg = eg
 
 		return nil
 	}
 }
 
+// WithInsertBufferSize returns the option to set the size of insert buffer.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/vqueue/option.go:48]:
```suggestion
// WithInsertBufferSize returns the option to set the size of the insert buffer.
```

File: pkg/agent/core/ngt/service/vqueue/option.go [hunk 2]
```diff
@@ -31,51 +33,61 @@ var defaultOptions = []Option{
...
+		if size <= 0 {
+			return errors.NewErrInvalidOption("insertBufferSize", size)
 		}
+		v.ichSize = size
 
 		return nil
 	}
 }
 
+// WithDeleteBufferSize returns the option to set the size of delete buffer.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/vqueue/option.go:60]:
```suggestion
// WithDeleteBufferSize returns the option to set the size of the delete buffer.
```

File: pkg/agent/core/ngt/service/vqueue/option.go [hunk 3]
```diff
@@ -31,51 +33,61 @@ var defaultOptions = []Option{
...
+		if size <= 0 {
+			return errors.NewErrInvalidOption("deleteBufferSize", size)
 		}
+		v.dchSize = size
 
 		return nil
 	}
 }
 
+// WithInsertBufferPoolSize returns the option to set the pool size of insert buffer.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/vqueue/option.go:72]:
```suggestion
// WithInsertBufferPoolSize returns the option to set the pool size of the insert buffer.
```

File: pkg/agent/core/ngt/service/vqueue/option.go [hunk 4]
```diff
@@ -31,51 +33,61 @@ var defaultOptions = []Option{
...
+		if size <= 0 {
+			return errors.NewErrInvalidOption("insertBufferPoolSize", size)
 		}
+		v.iBufSize = size
 
 		return nil
 	}
 }
 
+// WithDeleteBufferPoolSize returns the option to set the pool size of delete buffer.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/vqueue/option.go:84]:
```suggestion
// WithDeleteBufferPoolSize returns the option to set the pool size of the delete buffer.
```

File: pkg/agent/core/ngt/service/vqueue/option_test.go [hunk 5]
```diff
@@ -112,109 +94,87 @@ func TestWithErrGroup(t *testing.T) {
...
+				name: "set fails when size is -1",
+				args: args{
+					size: size,
+				},
+				want: want{
+					err: errors.NewErrInvalidOption("insertBufferSize", size),
+					obj: new(T),
+				},
+			}
+		}(),
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/vqueue/option_test.go:177]: could you add the test cases for checking the edge cases?

File: pkg/agent/core/ngt/service/vqueue/option_test.go [hunk 6]
```diff
@@ -229,109 +189,87 @@ func TestWithInsertBufferSize(t *testing.T) {
...
+				name: "set fails when size is 0",
+				args: args{
+					size: size,
+				},
+				want: want{
+					err: errors.NewErrInvalidOption("deleteBufferSize", size),
+					obj: new(T),
+				},
+			}
+		}(),
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/vqueue/option_test.go:272]: same as the above comment

File: pkg/agent/core/ngt/service/vqueue/option_test.go [hunk 7]
```diff
@@ -346,109 +284,87 @@ func TestWithDeleteBufferSize(t *testing.T) {
...
+				name: "set fails when size is 0",
+				args: args{
+					size: size,
+				},
+				want: want{
+					err: errors.NewErrInvalidOption("insertBufferPoolSize", size),
+					obj: new(T),
+				},
+			}
+		}(),
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/vqueue/option_test.go:367]: same as the above comment

File: pkg/agent/core/ngt/service/vqueue/option_test.go [hunk 8]
```diff
@@ -463,109 +379,87 @@ func TestWithInsertBufferPoolSize(t *testing.T) {
...
+				name: "set fails when size is 0",
+				args: args{
+					size: size,
+				},
+				want: want{
+					err: errors.NewErrInvalidOption("deleteBufferPoolSize", size),
+					obj: new(T),
+				},
+			}
+		}(),
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/vqueue/option_test.go:462]: same as the above comment

File: pkg/agent/core/ngt/service/vqueue/option_test.go (hunk 5 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/vqueue/option_test.go:177]: thank you
I fixed: https://github.com/vdaas/vald/pull/1233/commits/e7e9ced957228ab7bf6b9c6631c56597fb215595

File: pkg/agent/core/ngt/service/vqueue/option_test.go (hunk 6 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/vqueue/option_test.go]: thank you
I fixed: https://github.com/vdaas/vald/pull/1233/commits/e7e9ced957228ab7bf6b9c6631c56597fb215595

File: pkg/agent/core/ngt/service/vqueue/option_test.go (hunk 7 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/vqueue/option_test.go]: thank you
I fixed: https://github.com/vdaas/vald/pull/1233/commits/e7e9ced957228ab7bf6b9c6631c56597fb215595

File: pkg/agent/core/ngt/service/vqueue/option_test.go (hunk 8 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/vqueue/option_test.go]: thank you
I fixed: https://github.com/vdaas/vald/pull/1233/commits/e7e9ced957228ab7bf6b9c6631c56597fb215595

File: pkg/agent/core/ngt/service/vqueue/option_test.go (hunk 5 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/vqueue/option_test.go]: I fixed: https://github.com/vdaas/vald/pull/1233/commits/85883e379aba2cf52efde4c2ab12355015eeb322

File: pkg/agent/core/ngt/service/vqueue/option_test.go (hunk 6 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/vqueue/option_test.go]: I fixed: https://github.com/vdaas/vald/pull/1233/commits/85883e379aba2cf52efde4c2ab12355015eeb322

File: pkg/agent/core/ngt/service/vqueue/option_test.go (hunk 8 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/vqueue/option_test.go]: I fixed: https://github.com/vdaas/vald/pull/1233/commits/85883e379aba2cf52efde4c2ab12355015eeb322

**Comment by kevindiu** [comment]
Please also refactor https://github.com/vdaas/vald/blob/master/pkg/agent/core/ngt/service/vqueue/queue.go#L85  to handle option error, otherwise if the option is invalid, the ngt may not work due to your changes.

**Review by kevindiu — APPROVED** [review]
LGTM

**Review by vankichi — APPROVED** [review]
